source/postprocess/interface.cc

```diff
@@ -268,6 +268,101 @@ namespace aspect
+            pp = postprocessors.begin();
+            for (unsigned int i=0; i<postprocessor_names.size(); ++i, ++pp)
+              if (already_assigned[i] == false)
+                {
+                  // for this postprocessor, check if all of its dependencies
+                  // are already in the list
+                  const std::list<std::string> deps = (*pp)->required_other_postprocessors();
+                  bool unmet_dependencies = false;
+                  for (std::list<std::string>::const_iterator p = deps.begin();
+                       p != deps.end(); ++p)
+                    if (std::find (sorted_names.begin(),
+                                   sorted_names.end(),
+                                   *p) == sorted_names.end())
+                      {
+                        unmet_dependencies = true;
+                        break;
+                      }
+
+                  // if we have unmet dependencies, there is nothing we can do
+                  // right now for this postprocessor (but we will come back for it)
+                  //
+                  // if there are none, add this postprocessor
+                  if (unmet_dependencies == false)
+                    {
+                      sorted_names.push_back (postprocessor_names[i]);
+                      already_assigned[i] = true;
+                      at_least_one_element_added = true;
+                    }
+                }
+          }
+
+          // check that we have added at least one element; if not, there is a cycle
+          if (at_least_one_element_added == false)
+            {
+              std::ostringstream out;
+              out << "While sorting postprocessors by their dependencies, "
+                  "ASPECT encountered a cycle in dependencies. The following "
+                  "postprocessors are involved:\n";
+              typename std::vector<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+              pp = postprocessors.begin();
+              for (unsigned int i=0; i<postprocessor_names.size(); ++i, ++pp)
+                if (already_assigned[i] == false)
+                  {
+                    out << "  " << postprocessor_names[i] << " -> ";
+                    const std::list<std::string> deps = (*pp)->required_other_postprocessors();
+                    for (std::list<std::string>::const_iterator p = deps.begin();
+                         p != deps.end(); ++p)
+                      out << "'" << *p << "' ";
+                    out << std::endl;
+                  }
+              AssertThrow (false, ExcMessage(out.str()));
+            }
+        }
+      Assert (std::find (already_assigned.begin(), already_assigned.end(), false) == already_assigned.end(),
+              ExcInternalError());
+
+      // now that we have the sorted list of names, get the pointers to
+      // postprocessor in the same order
+      std::vector<std_cxx11::shared_ptr<Interface<dim> > > sorted_postprocessors;
```

Comment:
I suppose I could, but it would make the code above even more complicated. I will if you feel strongly about it. I thought it probably doesn't make much difference performancewise given that we do it only once at startup time.